viz_scripts/scaffolding.py

```diff
@@ -58,6 +58,13 @@ def load_all_confirmed_trips(tq):
     disp.display(all_ct.head())
     return all_ct
 
+def load_all_composite_trips(tq):
```

Comment:
we needed sections for a previous way that the survey choice was handled, but we should probably update them now, will the composite trips have the bluetooth sensed modes?

Reply:
The latest filtering for the survey assignment uses `confirmedMode?.baseMode == 'E_CAR'`, do you know where this confirmed mode comes from? I had been using the older version that relied on sections because that was what I had working (and therefore needed the composite trips). Is the bluetooth detected mode stored as a label?

Reply:
Yes, it will be once https://github.com/e-mission/e-mission-server/pull/965 is merged. In the interim, as a hack, we will need to store the `confirmedMode` in the public dashboard pre-processing by treating the BLE objects as user input.

Something like the following. 

```
import emission.storage.decorations.trip_queries as esdt

def get_confirmed_mode(ts, trip, ble_vehicle_mapping):
     matching_beacon = esdt.get_user_input_for_timeline_entry(ts, trip, "background/bluetooth_ble")
     return ble_vehicle_mapping.get(matching_beacon, None)
```

you can call it either before converting the trip list to a dataframe or to fill in the dataframe column later (preferable) using something like `df['confirmedMode'] = df.apply(lambda trip: get_confirmed_mode(ts, trip, ble_vehicle_mapping)`

Reply:
What are `ts` and `ble_vehicle_mapping`? I'm getting errors about `data` being missing from the trips as well, but this seemed to work out when I wrapped the trip like `{data: trip}` because I could see that it was searching for the attribute `data.start_ts`. I tried the `start_ts` first as `ts` thinking it was timestamp, but I think it's actually timeseries, so I tried `esta.TimeSeries.get_aggregate_time_series()` but I'm only getting `None` back

Reply:
@Abby-Wheelis you could use `get_user_input_for_trip` instead by passing in the `user_id` and `trip_id` from the dataframe (`trip_id` is `_id`) and the trip_key of `analysis/confirmed_trip`. It will create the timeseries for you and wrap the object and everything. It is not as efficient, but I also don't have the time to walk you through the e-mission data model at this time.

https://github.com/e-mission/e-mission-server/blob/bc2d7b6b5f9fa1a9e981bbc5b911d7be9aaa393e/emission/storage/decorations/trip_queries.py#L100

The BLE mapping is in the dynamic config.

Reply:
Yes, confirming that I am working on a more permanent solution. but for now you can compute `confirmedMode` in-place

I have some methods in emcommon.bluetooth for BLE matching so you won't have to re-implement too much

Reply:
I'm still getting `None` for all of the entries when I look up the `"background/bluetooth_ble"`, I see in the blame that this was added to the server about 3 weeks ago, which I'm imagining would have been right around the time we started the Alpha testing - and the data I have is from the 15th, which was chosen to have the data from when GSA was in town testing ... I was thinking maybe it wouldn't be in this data but I just talked myself out of that, I'll keep investigating I think this is something I must be doing and not the data

Reply:
I was using the dump from the 24th and the entries are there. don't know about the 15th

Reply:
I tried loading the most recent data, still no luck getting any of the `"background/bluetooth_ble"` entries. I am able to run the line `matching_beacon = esdt.get_user_input_for_trip("analysis/confirmed_trip", trip._id, trip.user_id, "background/bluetooth_ble")` across all of the trips, but am getting `None` in response for all of them. I have also tried with "cleaned_trip" and "composite_trip" but still all `None`, am I using the right trip key?

Reply:
Have now loaded the new data with `ble_sensed_summary` -- though not many of the trips seem to have a detected mode - of the 990 trips I have, 330 have an 'UNKOWN' mode, and 6 are sensed 'CAR'

I used `expanded_ct["confirmedMode?.baseMode"] = participant_ct_df.ble_sensed_summary.apply(lambda md: max(md["distance"], key=md["distance"].get))`, and it seems to be working well, but not sure if it's what we expect, I think I'd expect more trips sensed as `CAR`, and some sensed as `ECAR` ... maybe I need to reevaluate how we distill the mode?

Reply:
poking around at the entries for `ble_sensed_summary`, I do see some where `CAR` appears but is not the max by distance, is this expected? Are we picking up the beacon for some (but not all) of the trip? Or are they just trips with multiple segments? Only 6 trips with a detected beacon just seems quite low to me